crates/sui-core/src/consensus_handler.rs

```diff
@@ -2009,6 +2018,45 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                     // TODO(fastpath): Handle unlocking.
                     continue;
                 }
+                // When fastpath is disabled, perform post-consensus owned object conflict detection
+                // BEFORE setting Finalized status. If lock acquisition fails, the transaction has
+                // invalid/conflicting owned inputs and should be dropped.
+                if self.epoch_store.protocol_config().disable_fastpath()
+                    && let Some(tx) = parsed.transaction.kind.as_user_transaction()
+                    && let Ok(input_objects) = tx.transaction_data().input_objects()
+                {
+                    let owned_object_refs: Vec<_> = input_objects
+                        .iter()
+                        .filter_map(|obj| match obj {
+                            InputObjectKind::ImmOrOwnedMoveObject(obj_ref) => Some(*obj_ref),
+                            _ => None,
+                        })
+                        .collect();
+
+                    if !owned_object_refs.is_empty()
+                        && let Err(error) = self.cache_writer.acquire_transaction_locks(
```

Comment:
@mwtian caught an issue with this where we will incorrectly reject a transaction if execution is delayed. We need to ensure a new version of acquire_transaction_locks is used here that ALLOWS for future versions and not just the live version